pumpkin/src/command/commands/cmd_msg.rs

```diff
@@ -0,0 +1,69 @@
+use async_trait::async_trait;
+use pumpkin_core::text::{color::NamedColor, TextComponent};
+
+use crate::command::{
+    args::{arg_message::MsgArgConsumer, arg_players::PlayersArgumentConsumer, Arg, ConsumedArgs},
+    tree::CommandTree,
+    tree_builder::argument,
+    CommandExecutor, CommandSender, InvalidTreeError,
+};
+use InvalidTreeError::InvalidConsumptionError;
+
+const NAMES: [&str; 3] = ["msg", "tell", "w"];
+
+const DESCRIPTION: &str = "Sends a private message to one or more players.";
+
+const ARG_TARGET: &str = "target";
+const ARG_MESSAGE: &str = "message";
+
+struct MsgExecutor;
+
+#[async_trait]
+impl CommandExecutor for MsgExecutor {
+    async fn execute<'a>(
+        &self,
+        sender: &mut CommandSender<'a>,
+        _server: &crate::server::Server,
+        args: &ConsumedArgs<'a>,
+    ) -> Result<(), InvalidTreeError> {
+        let sender_name = match sender {
+            CommandSender::Console => "Console",
+            CommandSender::Rcon(_) => "Rcon",
+            CommandSender::Player(player) => &player.gameprofile.name.clone(),
+        };
+
+        let Some(Arg::Players(targets)) = args.get(&ARG_TARGET) else {
+            return Err(InvalidConsumptionError(Some(ARG_TARGET.into())));
+        };
+
+        let Some(Arg::Msg(msg)) = args.get(ARG_MESSAGE) else {
+            return Err(InvalidConsumptionError(Some(ARG_MESSAGE.into())));
+        };
```

Comment:
You should be able to just use something like ```let msg = MsgArgConsumer.find_arg(args, ARG_MESSAGE)?;```. Im pretty sure FindArg is implemented for all argument consumers, so same goes for PlayerArgConsumer

Reply:
Thanks, i'm using the `find_arg` now 🙏